src/platform_impl/windows/event_loop.rs

```diff
@@ -815,7 +840,10 @@ unsafe extern "system" fn public_window_callback<T: 'static>(
 
         winuser::WM_NCDESTROY => {
             drop(subclass_input);
-            Box::from_raw(subclass_input_ptr as *mut SubclassInput<T>);
+            remove_window_subclass(
+                window,
+                Box::from_raw(subclass_input_ptr as *mut SubclassInput<T>),
+            );
```

Comment:
Dropping subclass input first and then trying to do something with a pointer to it seems strange to me.

Reply:
The `subclass_input` that's dropped has a type of `&SubclassInput<T>` (introduced [here](https://github.com/rust-windowing/winit/blob/3e9681e8d94db0e72750b884fdb4ab0b3ab44d75/src/platform_impl/windows/event_loop.rs#L781)). Dropping it is necessary to avoid UB.

Reply:
> references are copy types

Ah, I had completely forgotten, but that then makes me wonder why it was ever there in the first place.

Reply:
I've dug into the code some more, and it seems like there isn't anything guaranteeing that the event target window gets destroyed after all other windows. This should be fixable by temporarily cloning and holding onto the `Rc` until we don't need it.
 The slightly more ambiguous issue is to do with pointer aliasing. It feels like having a `&` reference to something while also having a `*mut` pointer to it is insta-UB, but I'd have to consult with someone more knowledgeable on this.

Reply:
>  The slightly more ambiguous issue is to do with pointer aliasing. It feels like having a `&` reference to something while also having a `*mut` pointer to it is insta-UB, but I'll have to consult with someone more knowledgeable on this before reaching a conclusion.

I talked with someone on the rust community discord, and it seems like this is indeed UB.